src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -353,18 +411,25 @@ define([
       });
     },
 
-    showAppModal: function(app) {
+    routeApp: function(params) {
```

Comment:
Don't set state in a route function. this should be done in `componentDidMount`, i.e. before. This function should only return/result in a component.